src/geqrf_qdwh_full.cc

```diff
@@ -0,0 +1,490 @@
+{
+    int64_t lookahead = get_option<int64_t>( opts, Option::Lookahead, 1 );
+
+    int64_t ib = get_option<int64_t>( opts, Option::InnerBlocking, 16 );
+
+    int64_t max_panel_threads  = std::max(omp_get_max_threads()/2, 1);
+    max_panel_threads = get_option<int64_t>( opts, Option::MaxPanelThreads, max_panel_threads );
+
+    internal::specialization::geqrf_qdwh_full(internal::TargetType<target>(),
+                                    A, T,
+                                    ib, max_panel_threads, lookahead);
+}
+
+//------------------------------------------------------------------------------
+/// Distributed parallel QR factorization.
+///
+/// Computes a QR factorization of an m-by-n matrix $A$.
+/// The factorization has the form
+/// \[
+///     A = QR,
+/// \]
+/// where $Q$ is a matrix with orthonormal columns and $R$ is upper triangular
+/// (or upper trapezoidal if m < n).
+///
+/// Complexity (in real):
+/// - for $m \ge n$, $\approx 2 m n^{2} - \frac{2}{3} n^{3}$ flops;
+/// - for $m \le n$, $\approx 2 m^{2} n - \frac{2}{3} m^{3}$ flops;
+/// - for $m = n$,   $\approx \frac{4}{3} n^{3}$ flops.
+/// .
+//------------------------------------------------------------------------------
+/// @tparam scalar_t
+///     One of float, double, std::complex<float>, std::complex<double>.
+//------------------------------------------------------------------------------
+/// @param[in,out] A
+///     On entry, the m-by-n matrix $A$.
+///     On exit, the elements on and above the diagonal of the array contain
+///     the min(m,n)-by-n upper trapezoidal matrix $R$ (upper triangular
+///     if m >= n); the elements below the diagonal represent the unitary
+///     matrix $Q$ as a product of elementary reflectors.
+///
+/// @param[out] T
+///     On exit, triangular matrices of the block reflectors.
+///
+/// @param[in] opts
+///     Additional options, as map of name = value pairs. Possible options:
+///     - Option::Lookahead:
+///       Number of panels to overlap with matrix updates.
+///       lookahead >= 0. Default 1.
+///     - Option::InnerBlocking:
+///       Inner blocking to use for panel. Default 16.
+///     - Option::MaxPanelThreads:
+///       Number of threads to use for panel. Default omp_get_max_threads()/2.
+///     - Option::Target:
+///       Implementation to target. Possible values:
+///       - HostTask:  OpenMP tasks on CPU host [default].
+///       - HostNest:  nested OpenMP parallel for loop on CPU host.
+///       - HostBatch: batched BLAS on CPU host.
+///       - Devices:   batched BLAS on GPU device.
+///
+/// @ingroup geqrf_computational
+///
+template <typename scalar_t>
+void geqrf_qdwh_full(Matrix<scalar_t>& A,
+           TriangularFactors<scalar_t>& T,
+           Options const& opts)
+{
+    Target target = get_option( opts, Option::Target, Target::HostTask );
+
+    switch (target) {
+        case Target::Host:
+        case Target::HostTask:
+            geqrf_qdwh_full<Target::HostTask>(A, T, opts);
```

Comment:
Call `impl::geqrf_qdwh_full<Target::HostTask>( A, T, opts );`
Ideally with spaces inside the parens. Similarly in cases below. Cf. geqrf.cc.